### PR TITLE
crates-io: Add support for `db-dump.zip` files

### PR DESCRIPTION
**Pull request by Turbo87**
see https://github.com/rust-lang/crates.io/pull/8748

### CODE REVIEW

File: terragrunt/modules/crates-io/s3-static.tf [hunk 1]
```diff
@@ -27,6 +27,17 @@ resource "aws_s3_bucket" "static" {
...
+  lifecycle_rule {
+    id      = "purge-db-dump-zip"
+    enabled = true
+    prefix  = "db-dump.zip"
+
+    abort_incomplete_multipart_upload_days = 1
+    noncurrent_version_expiration {
+      days = 1
+    }
+  }
```
Comment (jdno) [inline terragrunt/modules/crates-io/s3-static.tf:39]: I'd like to avoid the duplication here. Can we use a [dynamic block](https://developer.hashicorp.com/terraform/language/expressions/dynamic-blocks) for this instead?

```hcl
dynamic "lifecycle_rule" {
  for_each = toset(["db-dump.tar.gz", "db-dump.zip"])

  content {
    id      = "purge-${replace(lifecycle_rule.key, ".", "-")}"
    enabled = true
    prefix  = lifecycle_rule.key

    abort_incomplete_multipart_upload_days = 1
    noncurrent_version_expiration {
      days = 1
    }
  }
}
```

**Review by jdno — CHANGES_REQUESTED** [review]

File: terragrunt/modules/crates-io/s3-static.tf (hunk 1 above)
Comment (Turbo87) [reply terragrunt/modules/crates-io/s3-static.tf:39]: > Can we use a [dynamic block](https://developer.hashicorp.com/terraform/language/expressions/dynamic-blocks) for this instead?

you tell me, I have no idea how to check if this works or not 😂

Comment (jdno) [reply terragrunt/modules/crates-io/s3-static.tf:39]: Sorry, let me rephrase the question. Can you please change it to a dynamic block? 😂 I already tested that the code above works.

Comment (Turbo87) [reply terragrunt/modules/crates-io/s3-static.tf:32]: done :)

**Review by jdno — APPROVED** [review]
This is live on staging. :shipit:
